streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -90,24 +91,61 @@ public void setup() {
         );
     }
 
+    private void startStreamsAndWaitForRunning(final KafkaStreams kafkaStreams) throws InterruptedException {
+        kafkaStreams.start();
+        waitForRunning();
+    }
+
     @After
     public void teardown() throws IOException {
+        stateTransitionHistory.clear();
```

Comment:
This is unnecessary as junit always create a new test class for each test case.

Reply:
Good suggestion!